im.pidgin.Pidgin

Pixelnarium opened the pull request:
This is a flatpak of the Pidgin Instant Messanger. I added 2 plugins I use myself. One is pidgin-otr which adds off-the-record encryption and the other one is pidgin-libnotify for desktop notifications. The last one needs some patches to work. I used the ones from the ArchLinux repositories.

I disabled some features because I don't know yet how to get them to work. i.e. I think NetworkManager support would need a NetworkManager installation, same with Avahi. It's pretty huge to start with.

Pidgin saves pretty much everything inside the ~/.purple folder.

TingPing commented on im.pidgin.Pidgin.json:
```diff
@@ -0,0 +1,134 @@
+    {
+      "name": "pidgin",
+      "buildsystem": "autotools",
+      "config-opts": ["--disable-vv", "--disable-meanwhile", "--disable-avahi", "--disable-nm", "--disable-tk"],
```
Please put these on their own lines.

TingPing commented on pidgin-libnotify-0.14-libnotify-0.7.patch:
```diff
@@ -0,0 +1,18 @@
+ 	notification = notify_notification_new (title, tr_body, NULL, NULL);
++#endif
++#else
++	notification = notify_notification_new (title, tr_body, NULL, NULL);
```
1. This patch is silly:

```c
#if !defined(NOTIFY_CHECK_VERSION) || !NOTIFY_CHECK_VERSION(0, 7, 0)
    notification = notify_notification_new (title, tr_body, NULL, NULL);
#else
    notification = notify_notification_new (title, tr_body, NULL, NULL);
#endif
```

2. Is this really not upstream, this version is ancient?

Pixelnarium replied:
It does not build without this patch. And pidgin-libnotify is quite old yes.

eszlari commented on appdata.xml:
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
The is already an AppStream metadata file in the upstream source tree:

tarball: `pidgin-2.13.0/pidgin/data/pidgin.appdata.xml.in`

repository:
https://bitbucket.org/pidgin/main/src/0b7a0190d83db173f80755c2d6267b0811cc07eb/pidgin/data/im.pidgin.Pidgin.appdata.xml.in

TingPing commented on im.pidgin.Pidgin.json:
```diff
@@ -0,0 +1,134 @@
+      ]
+    },
+    {
+      "name": "pidgin-otr",
```
We could put these in their own package. Unsure if its worth it. If upstream gets involved it would make sense to not bundle these.

Pixelnarium replied:
Is there an example of putting them in a seperate package?

eszlari commented on im.pidgin.Pidgin.json:
```diff
@@ -0,0 +1,134 @@
+    "--share=network",
+    "--socket=pulseaudio",
+    "--persist=.purple",
+    "--filesystem=xdg-download:rw",
```
"rw" can be dropped.

eszlari commented on im.pidgin.Pidgin.json:
```diff
@@ -0,0 +1,134 @@
+    },
+    {
+      "name": "pidgin",
+      "buildsystem": "autotools",
```
Can be dropped (autotools is the default).

TingPing commented on appdata.xml:
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>im.pidgin.Pidgin</id>
+  <launchable type="desktop-id">im.pidgin.Pidgin</launchable>
```
Missing `.desktop` on the end.

TingPing commented:
Could you contact upstream to see if they are interested in this.

TingPing commented:
bot, build im.pidgin.Pidgin

eszlari commented:
Have you considered using YAML instead of JSON?

Pixelnarium commented:
@eszlari 

I am not that experienced with YAML so it was pretty hard for me to adapt a JSON example to YAML. So I am just using JSON for now. Learn how flatpak works is enough for now.

@TingPing OK, I will search for a project that does that.

eszlari commented:
> I am not that experienced with YAML so it was pretty hard for me to adapt a JSON example to YAML.

https://www.json2yaml.com
https://github.com/flathub/flathub/wiki/YAML-Style-Guide

barthalion commented:
What's with this YAML campaigning? JSON isn't any worse really, it's all up to the submitter.

Pixelnarium commented:
OK, I tried the autoconf regeneration. I think it is `"rm-configure": true` but this does not work because pidgin omits the autogen.sh from the source.tar. So flatpak complains `Error: module pidgin: Can't find autogen, autogen.sh or bootstrap`.

I also can't pull from the source repository because flatpak does not support Mercurial.

Side node: HEAD switched from autotools to meson. Maybe it's part of the next release, maybe not.

TingPing commented:
> but this does not work because pidgin omits the autogen.sh from the source.tar. 

Just make your own, random example: https://github.com/flathub/shared-modules/blob/e2374d4cde51f312f2c39401d59eb4248fe60449/libappindicator/libappindicator-gtk3-introspection-12.10.json#L40-L44

eszlari commented:
> What's with this YAML campaigning? JSON isn't any worse really, it's all up to the submitter.

Obviously, it's up to the maintainer. But not everyone is aware that this option even exists (see e.g. https://github.com/flathub/flathub/pull/551).

AsciiWolf commented:
[Failed](https://flathub.org/builds/#/builders/37/builds/322) to build on aarch64.

nedrichards commented:
@Pixelnarium did you also have an autogen.sh script module like so: https://github.com/flathub/shared-modules/blob/08e9b9eead7a95139be5d1fbc24bea730fb78f7a/SDL/SDL_net-1.2.8.json ? You need that as well as rm-configure.

TingPing commented:
bot, build im.pidgin.Pidgin

nedrichards commented:
It does indeed build: https://flathub.org/builds/#/builders/2/builds/3617

Pixelnarium commented:
So, I changed some things. First I removed the plugins since they are probably abandoned and I could not get the flatpak extension system to work due to the fact that the plugins are not just single .so files. This could be revisited later if there is some need.

I added farstream and avahi as dependency for some Pidgin features.

nedrichards commented:
bot, build im.pidgin.Pidgin

nedrichards commented:
Huh? Extensions are for directories, not bare files aren't they? like so: https://github.com/flathub/org.pitivi.Pitivi/blob/master/org.pitivi.Pitivi.json#L37

Pixelnarium commented:
There are also problems with the installation scripts of the plugins. But to name a few things:
- plugins install localization files, I don't know how this is handled
- the install scripts read the plugin directory from the pidgin/libpurple installation. This is /app/lib/purple-2 if I remember correctly. But this directory is read-only for plugin-installations since it belongs to the pidgin "runtime". And I could not figure out how to change this. But this is only the path for the plugin itself and not everything else (things that would normally land in /app/bin, /app/lib, /app/share)
- due to so few examples out there this this is mostly try&error on my side

I am not giving up on this but I feel like bundling things only I use is not a good "standard" package. So I removed them for now.

nedrichards commented:
Oh right, yeah - you need to be able to confine everything to a directory (and set of subdirectories) rather than plugins expecting to scatter files throughout the system.

Pixelnarium commented:
Just me for now.

nedrichards commented:
Repository has been created: https://github.com/flathub/im.pidgin.Pidgin
